File: C-Language/quicksort.c
#include <stdio.h>

#define SWAP(x, y, t) { t = x; x = y; y = t; }

void quicksort(int list[], int left, int right) {
    int pivot, i, j, temp;
    if (left < right) {
        pivot = left;
        i = left;
        j = right;
        while (i < j) {
            while (list[i] <= list[pivot] && i < right)
                i++;
            while (list[j] > list[pivot])
                j--;
            if (i < j) {
                SWAP(list[i], list[j], temp);
            }
        }
        SWAP(list[pivot], list[j], temp);
        quicksort(list, left, j - 1);
        quicksort(list, j + 1, right);
    }
}

int main() {
    int list[] = {34, 7, 23, 32, 5, 62};
    int n = sizeof(list) / sizeof(list[0]);
    quicksort(list, 0, n - 1);
    for (int i = 0; i < n; i++) {
        printf("%d ", list[i]);
    }
    return 0;
}

File: Python/quicksort.py
def quicksort(arr):
  if len(arr) <= 1:
    return arr

  pivot = arr[len(arr) // 2]
  left = [x for x in arr if x < pivot]
  middle = [x for x in arr if x == pivot]
  right = [x for x in arr if x > pivot]

  return quicksort(left) + middle + quicksort(right)

if __name__ == "__main__":
  list1 = [34, 7, 23, 32, 5, 62]
  print("input array:",list1)
  sorted_list = quicksort(list1)
  print("output array:",sorted_list)
